salt/metalk8s/orchestrate/upgrade/etcd.sls

```diff
@@ -0,0 +1,52 @@
+{%- set dest_version = pillar.orchestrate.dest_version %}
+{%- set etcd_nodes = salt.metalk8s.minions_by_role('etcd') %}
+{%- set control_plane_ip = salt.saltutil.runner(
+    'mine.get',
+    tgt=pillar.bootstrap_id,
+    fun='control_plane_ip'
+)[pillar.bootstrap_id] %}
+
+{%- set pillar_data = {
+    'repo': {
+        'host': control_plane_ip
+    },
+    'registry_ip': control_plane_ip
+} %}
+
+Check etcd cluster health:
+  module.run:
+    - metalk8s.check_etcd_health:
+      - hostname: {{ pillar.bootstrap_id }}
+    - failhard: True
+
+{%- for node in etcd_nodes %}
+
+Sync {{ node }} minion:
+  salt.function:
+    - name: saltutil.sync_all
+    - tgt: {{ node }}
+    - saltenv: metalk8s-{{ dest_version }}
+
+Upgrade etcd {{ node }} to {{ dest_version }}:
+  salt.state:
+    - tgt: {{ node }}
+    - sls:
+      - metalk8s.kubernetes.etcd.health
+    - saltenv: metalk8s-{{ dest_version }}
+    - pillar: {{ pillar_data | tojson }}
+    - require:
+      - salt: Sync {{ node }} minion
+
+Check etcd cluster health for {{ node }}:
+  module.run:
+    - metalk8s.check_etcd_health:
+      - hostname: {{ node }}
+    - failhard: True
```

Comment:
Same comment.

Reply:
See above :)

Reply:
Still not overly happy with it :P

Reply:
Like I said in the other comment,

Upgrade of one node not really require an upgrade of the previous node.

Most of the case for me `require`-chain is better but for rolling upgrade a `failhard` it's something good no ?
"I'm not able to upgrade one node lets stop here" that make sense no ?

Reply:
I think I agree with @TeddyAndrieux here: there is no "real" dependency between etcd node upgrades, they are (supposed to be) interchangeable, IIUC. Since it's just semantically about "run this batch in sequence, stop at first failure", I think `failhard: True` makes more sense.

Reply:
As discussed directly with Nicolas and Guillaume let's do `require` chain to have a clean ordering and maybe do something smarter in the futur

Reply:
Done